pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -271,7 +283,13 @@ func (r *repository) Get(ctx context.Context, dgst digest.Digest, options ...dis
 	}
 
 	ref := imageapi.DockerImageReference{Namespace: r.namespace, Name: r.name, Registry: r.registryAddr}
-	manifest, err := r.manifestFromImageWithCachedLayers(image, ref.DockerClientDefaults().Exact())
+	if managed := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; managed == "true" {
+		ref.Registry = ""
```

Comment:
> should we really be clearing this, or setting it to a "local registry" value? the effects of this change are very hard to reason about

There are two kind of entries in the `cachedLayers`:

1. `<registry>/<namespace>/<name>`
2. `<namespace>/<name>`

The former is set and used by a pullthrough middleware which operates on remote blobs, it's not interested in the latter.

The latter is set and used by blob descriptor service operating on local blobs. It's also set on this place when it's clear that image's blobs are or should be available locally. Registry's location might change over time so I'd rather avoid depending on it.

Nevertheless the code for not managed image is wrong anyway (even prior to this change). The code will append `:latest` tag causing cache misses in the pullthrough middleware. I'll fix it.